Content.Server/Nutrition/EntitySystems/FoodSystem.cs

```diff
@@ -95,6 +97,20 @@ private void OnFeedFood(EntityUid uid, FoodComponent foodComponent, AfterInterac
             if (food == user || TryComp<MobStateComponent>(food, out var mobState) && _mobStateSystem.IsAlive(food, mobState)) // Suppresses eating alive mobs
                 return (false, false);
 
+            // Check if there's a required tag and if it 
+            if (foodComp.RequiredTag != String.Empty && !_tags.HasTag(target, foodComp.RequiredTag))
+                return (false, false);
+
+            Console.Write(foodComp.RequiredTag != String.Empty && !_tags.HasTag(target, foodComp.RequiredTag));
+            Console.Write(_tags.HasTag(target, "ClothEater") && (foodComp.RequiredTag == String.Empty || foodComp.RequiredTag != "ClothEater"));
+
+            // Make sure moths can't eat anything else
+            if (_tags.HasTag(target, "ClothEater") && (foodComp.RequiredTag == String.Empty || foodComp.RequiredTag != "ClothEater"))
+            {
+                _popupSystem.PopupEntity(Loc.GetString("food-repulsing"), target);
+                return (false, false);
+            }
+
```

Comment:
Don't hardcode checks in here, use events and another system as mentioned above.

Also do not use Console.Write, use the logger.